src/main.rs

```diff
@@ -85,115 +78,32 @@ fn main() -> CliResult {
         }
     };
 
-    // if let Some(self::commands::CliTopLevelCommand::GenerateShellCompletions(subcommand)) =
-    //     cli.top_level_command
-    // {
-    //     subcommand.process();
-    //     return Ok(());
-    // }
-
-    let cmd = loop {
-        match Cmd::from_cli(Some(cli.clone()), (config.clone(),)) {
-            Ok(Some(cmd)) => {
-                break cmd;
+    loop {
+        match <Cmd as interactive_clap::FromCli>::from_cli(Some(cli.clone()), (config.clone(),)) {
+            interactive_clap::ResultFromCli::Ok(cli_cmd)
+            | interactive_clap::ResultFromCli::Cancel(Some(cli_cmd)) => {
+                println!(
+                    "Your console command:\n{} {}",
+                    std::env::args().next().as_deref().unwrap_or("./near_cli"),
+                    shell_words::join(&cli_cmd.to_cli_args())
+                );
+                return Ok(());
             }
-            Ok(None) => {}
-            Err(err) => match err.downcast_ref() {
-                Some(
-                    inquire::InquireError::OperationCanceled
-                    | inquire::InquireError::OperationInterrupted,
-                ) => {
-                    println!("<Operation was interrupted. Goodbye>");
-                    return Ok(());
-                }
-                Some(_) | None => return Err(err),
-            },
-        }
-    };
-
-    let completed_cli = CliCmd::from(cmd.clone());
-
-    let process_result = tokio::runtime::Runtime::new()
-        .unwrap()
-        .block_on(cmd.process(config));
-
-    println!(
-        "Your console command:\n{} {}",
-        std::env::args().next().as_deref().unwrap_or("./near_cli"),
-        shell_words::join(completed_cli.to_cli_args())
-    );
-
-    #[cfg(feature = "self-update")]
-    {
-        if !matches!(
-            cmd,
-            Cmd {
-                top_level: crate::commands::TopLevelCommand::Extensions(
-                    crate::commands::extensions::ExtensionsCommands {
-                        extensions_actions:
-                            crate::commands::extensions::ExtensionsActions::SelfUpdate(
-                                crate::commands::extensions::self_update::SelfUpdateCommand,
-                            ),
-                    },
-                ),
```

Comment:
Please, don't remove `self-update` functionality that was recently added in #165.

Reply:
done